### DISCUSSION TITLE
Image drag/drop isn't working with latest beta? 1.3.0-beta.20

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#332] [Errors] [answered]
The specific script: https://github.com/johnlindquist/kit/discussions/310

![no-worky](https://user-images.githubusercontent.com/1500684/123708809-42beee80-d829-11eb-8db2-b2ba47651e50.gif)

Doesn't seem to work in the terminal either (is it supposed to?):

```
~ 💥
$ cloudinary-upload 
✔ Select directory in kentcdodds.com/misc · .
✔ Drop the image(s) you want to upload · /Users/kentcdodds/Desktop/alex.png 
The "path" argument must be of type string. Received undefined
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:329:5)
    at validateString (node:internal/validators:129:11)
    at Object.parse (node:path:1327:5)
    at file:///Users/kentcdodds/.kenv/scripts/cloudinary-upload.js?uuid=e399e2c3-5f97-40a1-84ef-34e0c5f59de7:100:28
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:329:5)
    at validateString (node:internal/validators:129:11)
    at Object.parse (node:path:1327:5)
    at file:///Users/kentcdodds/.kenv/scripts/cloudinary-upload.js?uuid=e399e2c3-5f97-40a1-84ef-34e0c5f59de7:100:28
    at processTicksAndRejections (node:internal/process/task_queues:94:5) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

It's been a few weeks since I tried this so I'm not sure when it broke.

**Comment by johnlindquist** [comment, marked as answer]
Please try replacing:

```js
const images = await arg({
  placeholder: 'Drop the image(s) you want to upload',
  drop: true,
  ignoreBlur: true,
})
```

with:

```js
const images = await drop('Drop the image(s) you want to upload')
```

## Why

`drop:true` was removed from the `arg` options since more UI options are being added. `drop` is really just a helper that wraps `arg`. If you want to use the `arg` version for whatever reason, it would be:

```js
const images = await arg({
  placeholder: 'Drop the image(s) you want to upload',
  ignoreBlur: true,
  ui: 'drop',
})
```

The other ui helpers:
* `await hotkey()`
* `await textarea()`
* `await editor()`

all wrap `arg` in a similar fashion.

**Reply by johnlindquist** [reply]
Oh, and supporting the various ui helpers in the terminal is very low on the list of priorities. I do need to add "unsupported in the terminal" or some other messaging if someonetries.

**Reply by kentcdodds** [reply]
Sweet, that fixed it. Thanks!

**Reply by JamesHenry** [reply]
@johnlindquist For me on beta.20 the above two commands are not equivalent, as the `drop()` variant does not seem to apply the placeholder.

Concretely, in my case:

```ts
const dropData = await arg({
  placeholder: 'Drop all account statement.csv file(s)',
  ignoreBlur: true,
  ui: 'drop',
});
```

...correctly renders:

![image](https://user-images.githubusercontent.com/900523/124354892-7762e980-dc1f-11eb-8d81-8325927ee234.png)

Whereas:

```ts
const dropData = await drop('Drop all account statement.csv file(s)');
```

...incorrectly (based on my interpretation of your comment) renders:

![image](https://user-images.githubusercontent.com/900523/124354907-8c3f7d00-dc1f-11eb-8016-c9ba57f58754.png)

P.S. You will see in both cases that the bottom border is missing. It doesn't look like it would be intentional, is that visual issue on your radar?

**Reply by johnlindquist** [reply]
@JamesHenry Thanks! I'll get those issues sorted.

**Reply by JamesHenry** [reply]
@johnlindquist I've been keeping an eye out for a release entry on Discussions, but see we're actually on `.23` already.

I can confirm that the placeholder text is now being respected and showing correctly so that's great, however the visual aspect is now arguably worse than before. The bottom border still does not render, and the drop area has additionally got considerably smaller.

(I didn't change anything to do with any of the scripts)

![image](https://user-images.githubusercontent.com/900523/125305698-9ceaa400-e33f-11eb-9a1c-7239806f764c.png)

If you would like me to open a separate issue or discussion to track this please let me know

**Reply by johnlindquist** [reply]
Here is fine. I wasn't testing with the //Menu metadata and was simply hard-coding the height. Obviously that won't work. 😅

**Reply by JamesHenry** [reply]
Things looking good in `.29`, thanks @johnlindquist!
